File: accepted-solutions/spavanac.cpp
#include <iostream>
using namespace std;

int main()
{
    int hours, min;

    cin >> hours;
    cin >> min;

    min -= 45;

    if (min < 0)
    {
        if (hours == 0)
            hours = 23;
        else
            --hours;

        min += 60;
    }    

    cout << hours << " " << min << endl;

    return 0;
}

File: accepted-solutions/spavanac/spavanac.java
import java.util.Scanner;

public class spavanac {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int hours = scan.nextInt();
        int min = scan.nextInt() - 45;

        if (min < 0) {
            if (hours == 0) {
                hours = 23;
            }
            else {
                --hours;
            }

            min += 60;
        }

        System.out.println(hours + " " + min);
    }
}
